provider/godaddy/client.go

```diff
@@ -240,9 +259,9 @@ func (c *Client) Do(req *http.Request) (*http.Response, error) {
 
 		c.Ratelimiter.Wait(req.Context())
 		resp, err = c.Client.Do(req)
-	}
-	if err != nil {
-		return nil, err
+		if err != nil {
+			return nil, fmt.Errorf("doing request after waiting for retry after: %w", err)
+		}
 	}
```

Comment:
ust an idea. If the service tried 3 times, and still response is 429, probably better to return an error, It could be that implicit error handling make it clear that the request failed due to throttling, which will prevent unnecessary resource consumption by avoiding unmarshalling object and etc.

Something like before the return statement
```
if resp != nil && resp.StatusCode == 429 {
		return nil, fmt.Errorf("rate limit exceeded after X attempts")
	}
```

Reply:
> which will prevent unnecessary resource consumption by avoiding unmarshalling object and etc.

I don't see it as unnecessary, the response may contain more information as to why the request is still being rate limited even after waiting to retry. The resource consumption for unmarshalling the very small response from the API is also negligible IMO.